// xendit-node/refund/models/CreateRefund409Response.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateRefund409ResponseToJSON = exports.CreateRefund409ResponseFromJSONTyped = exports.CreateRefund409ResponseFromJSON = exports.instanceOfCreateRefund409Response = exports.CreateRefund409ResponseErrorCodeEnum = void 0;
const runtime_1 = require("../../runtime");
/**
 * @export
 */
exports.CreateRefund409ResponseErrorCodeEnum = {
    DuplicateError: 'DUPLICATE_ERROR',
    IdempotencyError: 'IDEMPOTENCY_ERROR'
};
/**
 * Check if a given object implements the CreateRefund409Response interface.
 */
function instanceOfCreateRefund409Response(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfCreateRefund409Response = instanceOfCreateRefund409Response;
function CreateRefund409ResponseFromJSON(json) {
    return CreateRefund409ResponseFromJSONTyped(json, false);
}
exports.CreateRefund409ResponseFromJSON = CreateRefund409ResponseFromJSON;
function CreateRefund409ResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'errorCode': !(0, runtime_1.exists)(json, 'error_code') ? undefined : json['error_code'],
        'message': !(0, runtime_1.exists)(json, 'message') ? undefined : json['message'],
    };
}
exports.CreateRefund409ResponseFromJSONTyped = CreateRefund409ResponseFromJSONTyped;
function CreateRefund409ResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'error_code': value.errorCode,
        'message': value.message,
    };
}
exports.CreateRefund409ResponseToJSON = CreateRefund409ResponseToJSON;
