// xendit-node/refund/models/CreateRefund503Response.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateRefund503ResponseToJSON = exports.CreateRefund503ResponseFromJSONTyped = exports.CreateRefund503ResponseFromJSON = exports.instanceOfCreateRefund503Response = exports.CreateRefund503ResponseErrorCodeEnum = void 0;
const runtime_1 = require("../../runtime");
/**
 * @export
 */
exports.CreateRefund503ResponseErrorCodeEnum = {
    ChannelUnavailable: 'CHANNEL_UNAVAILABLE',
    OtpDeliveryError: 'OTP_DELIVERY_ERROR'
};
/**
 * Check if a given object implements the CreateRefund503Response interface.
 */
function instanceOfCreateRefund503Response(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfCreateRefund503Response = instanceOfCreateRefund503Response;
function CreateRefund503ResponseFromJSON(json) {
    return CreateRefund503ResponseFromJSONTyped(json, false);
}
exports.CreateRefund503ResponseFromJSON = CreateRefund503ResponseFromJSON;
function CreateRefund503ResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'errorCode': !(0, runtime_1.exists)(json, 'error_code') ? undefined : json['error_code'],
        'message': !(0, runtime_1.exists)(json, 'message') ? undefined : json['message'],
    };
}
exports.CreateRefund503ResponseFromJSONTyped = CreateRefund503ResponseFromJSONTyped;
function CreateRefund503ResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'error_code': value.errorCode,
        'message': value.message,
    };
}
exports.CreateRefund503ResponseToJSON = CreateRefund503ResponseToJSON;
