// xendit-node/refund/models/CreateRefund403Response.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateRefund403ResponseToJSON = exports.CreateRefund403ResponseFromJSONTyped = exports.CreateRefund403ResponseFromJSON = exports.instanceOfCreateRefund403Response = exports.CreateRefund403ResponseErrorCodeEnum = void 0;
const runtime_1 = require("../../runtime");
/**
 * @export
 */
exports.CreateRefund403ResponseErrorCodeEnum = {
    Unauthorized: 'UNAUTHORIZED'
};
/**
 * Check if a given object implements the CreateRefund403Response interface.
 */
function instanceOfCreateRefund403Response(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfCreateRefund403Response = instanceOfCreateRefund403Response;
function CreateRefund403ResponseFromJSON(json) {
    return CreateRefund403ResponseFromJSONTyped(json, false);
}
exports.CreateRefund403ResponseFromJSON = CreateRefund403ResponseFromJSON;
function CreateRefund403ResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'errorCode': !(0, runtime_1.exists)(json, 'error_code') ? undefined : json['error_code'],
        'message': !(0, runtime_1.exists)(json, 'message') ? undefined : json['message'],
    };
}
exports.CreateRefund403ResponseFromJSONTyped = CreateRefund403ResponseFromJSONTyped;
function CreateRefund403ResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'error_code': value.errorCode,
        'message': value.message,
    };
}
exports.CreateRefund403ResponseToJSON = CreateRefund403ResponseToJSON;
