// xendit-node/refund/models/CreateRefund404Response.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateRefund404ResponseToJSON = exports.CreateRefund404ResponseFromJSONTyped = exports.CreateRefund404ResponseFromJSON = exports.instanceOfCreateRefund404Response = exports.CreateRefund404ResponseErrorCodeEnum = void 0;
const runtime_1 = require("../../runtime");
/**
 * @export
 */
exports.CreateRefund404ResponseErrorCodeEnum = {
    DataNotFound: 'DATA_NOT_FOUND'
};
/**
 * Check if a given object implements the CreateRefund404Response interface.
 */
function instanceOfCreateRefund404Response(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfCreateRefund404Response = instanceOfCreateRefund404Response;
function CreateRefund404ResponseFromJSON(json) {
    return CreateRefund404ResponseFromJSONTyped(json, false);
}
exports.CreateRefund404ResponseFromJSON = CreateRefund404ResponseFromJSON;
function CreateRefund404ResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'errorCode': !(0, runtime_1.exists)(json, 'error_code') ? undefined : json['error_code'],
        'message': !(0, runtime_1.exists)(json, 'message') ? undefined : json['message'],
    };
}
exports.CreateRefund404ResponseFromJSONTyped = CreateRefund404ResponseFromJSONTyped;
function CreateRefund404ResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'error_code': value.errorCode,
        'message': value.message,
    };
}
exports.CreateRefund404ResponseToJSON = CreateRefund404ResponseToJSON;
