// xendit-node/invoice/models/BankCode.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BankCodeToJSON = exports.BankCodeFromJSONTyped = exports.BankCodeFromJSON = exports.BankCode = void 0;
/**
 * Representing the available bank channels used for invoice-related transactions.
 * @export
 */
exports.BankCode = {
    Bca: 'BCA',
    Bni: 'BNI',
    Bri: 'BRI',
    Mandiri: 'MANDIRI',
    Permata: 'PERMATA',
    Bsi: 'BSI',
    Bjb: 'BJB',
    SahabatSampoerna: 'SAHABAT_SAMPOERNA',
    Cimb: 'CIMB',
    Bnc: 'BNC',
    Hana: 'HANA',
    Muamalat: 'MUAMALAT',
    XenditEnumDefaultFallback: "UNKNOWN_ENUM_VALUE"
};
function BankCodeFromJSON(json) {
    return BankCodeFromJSONTyped(json, false);
}
exports.BankCodeFromJSON = BankCodeFromJSON;
function BankCodeFromJSONTyped(json, ignoreDiscriminator) {
    if (json !== "" || json !== null) {
        const key = Object.keys(exports.BankCode)[Object.values(exports.BankCode).indexOf(json)];
        return exports.BankCode[key] === undefined ?
            exports.BankCode['XenditEnumDefaultFallback'] : exports.BankCode[key];
    }
    return json;
}
exports.BankCodeFromJSONTyped = BankCodeFromJSONTyped;
function BankCodeToJSON(value) {
    return value;
}
exports.BankCodeToJSON = BankCodeToJSON;
