// xendit-node/invoice/models/EwalletType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EwalletTypeToJSON = exports.EwalletTypeFromJSONTyped = exports.EwalletTypeFromJSON = exports.EwalletType = void 0;
/**
 * Representing the available eWallet channels used for invoice-related transactions.
 * @export
 */
exports.EwalletType = {
    Ovo: 'OVO',
    Dana: 'DANA',
    Linkaja: 'LINKAJA',
    Paymaya: 'PAYMAYA',
    Shopeepay: 'SHOPEEPAY',
    Gcash: 'GCASH',
    Grabpay: 'GRABPAY',
    Astrapay: 'ASTRAPAY',
    Nexcash: 'NEXCASH',
    Jeniuspay: 'JENIUSPAY',
    Momo: 'MOMO',
    Zalopay: 'ZALOPAY',
    Viettelpay: 'VIETTELPAY',
    Vnptwallet: 'VNPTWALLET',
    Appota: 'APPOTA',
    Linepay: 'LINEPAY',
    Truemoney: 'TRUEMONEY',
    Wechatpay: 'WECHATPAY',
    Touchngo: 'TOUCHNGO',
    XenditEnumDefaultFallback: "UNKNOWN_ENUM_VALUE"
};
function EwalletTypeFromJSON(json) {
    return EwalletTypeFromJSONTyped(json, false);
}
exports.EwalletTypeFromJSON = EwalletTypeFromJSON;
function EwalletTypeFromJSONTyped(json, ignoreDiscriminator) {
    if (json !== "" || json !== null) {
        const key = Object.keys(exports.EwalletType)[Object.values(exports.EwalletType).indexOf(json)];
        return exports.EwalletType[key] === undefined ?
            exports.EwalletType['XenditEnumDefaultFallback'] : exports.EwalletType[key];
    }
    return json;
}
exports.EwalletTypeFromJSONTyped = EwalletTypeFromJSONTyped;
function EwalletTypeToJSON(value) {
    return value;
}
exports.EwalletTypeToJSON = EwalletTypeToJSON;
