// xendit-node/invoice/models/ForbiddenError.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ForbiddenErrorToJSON = exports.ForbiddenErrorFromJSONTyped = exports.ForbiddenErrorFromJSON = exports.instanceOfForbiddenError = exports.ForbiddenErrorErrorCodeEnum = void 0;
/**
 * @export
 */
exports.ForbiddenErrorErrorCodeEnum = {
    AccessSuspended: 'ACCESS_SUSPENDED'
};
/**
 * Check if a given object implements the ForbiddenError interface.
 */
function instanceOfForbiddenError(value) {
    let isInstance = true;
    isInstance = isInstance && "errorCode" in value;
    isInstance = isInstance && "message" in value;
    return isInstance;
}
exports.instanceOfForbiddenError = instanceOfForbiddenError;
function ForbiddenErrorFromJSON(json) {
    return ForbiddenErrorFromJSONTyped(json, false);
}
exports.ForbiddenErrorFromJSON = ForbiddenErrorFromJSON;
function ForbiddenErrorFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'errorCode': json['error_code'],
        'message': json['message'],
    };
}
exports.ForbiddenErrorFromJSONTyped = ForbiddenErrorFromJSONTyped;
function ForbiddenErrorToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'error_code': value.errorCode,
        'message': value.message,
    };
}
exports.ForbiddenErrorToJSON = ForbiddenErrorToJSON;
