// xendit-node/invoice/models/InvoiceCurrency.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvoiceCurrencyToJSON = exports.InvoiceCurrencyFromJSONTyped = exports.InvoiceCurrencyFromJSON = exports.InvoiceCurrency = void 0;
/**
 * Representing the currency used for an invoice.
 * @export
 */
exports.InvoiceCurrency = {
    Idr: 'IDR',
    Usd: 'USD',
    Thb: 'THB',
    Vnd: 'VND',
    Php: 'PHP',
    XenditEnumDefaultFallback: "UNKNOWN_ENUM_VALUE"
};
function InvoiceCurrencyFromJSON(json) {
    return InvoiceCurrencyFromJSONTyped(json, false);
}
exports.InvoiceCurrencyFromJSON = InvoiceCurrencyFromJSON;
function InvoiceCurrencyFromJSONTyped(json, ignoreDiscriminator) {
    if (json !== "" || json !== null) {
        const key = Object.keys(exports.InvoiceCurrency)[Object.values(exports.InvoiceCurrency).indexOf(json)];
        return exports.InvoiceCurrency[key] === undefined ?
            exports.InvoiceCurrency['XenditEnumDefaultFallback'] : exports.InvoiceCurrency[key];
    }
    return json;
}
exports.InvoiceCurrencyFromJSONTyped = InvoiceCurrencyFromJSONTyped;
function InvoiceCurrencyToJSON(value) {
    return value;
}
exports.InvoiceCurrencyToJSON = InvoiceCurrencyToJSON;
