// xendit-node/invoice/models/InvoiceError404ResponseDefinition.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvoiceError404ResponseDefinitionToJSON = exports.InvoiceError404ResponseDefinitionFromJSONTyped = exports.InvoiceError404ResponseDefinitionFromJSON = exports.instanceOfInvoiceError404ResponseDefinition = exports.InvoiceError404ResponseDefinitionErrorCodeEnum = void 0;
/**
 * @export
 */
exports.InvoiceError404ResponseDefinitionErrorCodeEnum = {
    InvoiceNotFoundError: 'INVOICE_NOT_FOUND_ERROR'
};
/**
 * Check if a given object implements the InvoiceError404ResponseDefinition interface.
 */
function instanceOfInvoiceError404ResponseDefinition(value) {
    let isInstance = true;
    isInstance = isInstance && "errorCode" in value;
    isInstance = isInstance && "message" in value;
    return isInstance;
}
exports.instanceOfInvoiceError404ResponseDefinition = instanceOfInvoiceError404ResponseDefinition;
function InvoiceError404ResponseDefinitionFromJSON(json) {
    return InvoiceError404ResponseDefinitionFromJSONTyped(json, false);
}
exports.InvoiceError404ResponseDefinitionFromJSON = InvoiceError404ResponseDefinitionFromJSON;
function InvoiceError404ResponseDefinitionFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'errorCode': json['error_code'],
        'message': json['message'],
    };
}
exports.InvoiceError404ResponseDefinitionFromJSONTyped = InvoiceError404ResponseDefinitionFromJSONTyped;
function InvoiceError404ResponseDefinitionToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'error_code': value.errorCode,
        'message': value.message,
    };
}
exports.InvoiceError404ResponseDefinitionToJSON = InvoiceError404ResponseDefinitionToJSON;
