// xendit-node/invoice/models/PaylaterType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaylaterTypeToJSON = exports.PaylaterTypeFromJSONTyped = exports.PaylaterTypeFromJSON = exports.PaylaterType = void 0;
/**
 * Representing the available paylater channels used for invoice-related transactions.
 * @export
 */
exports.PaylaterType = {
    Kredivo: 'KREDIVO',
    Akulaku: 'AKULAKU',
    Uangme: 'UANGME',
    Billease: 'BILLEASE',
    Cashalo: 'CASHALO',
    Atome: 'ATOME',
    XenditEnumDefaultFallback: "UNKNOWN_ENUM_VALUE"
};
function PaylaterTypeFromJSON(json) {
    return PaylaterTypeFromJSONTyped(json, false);
}
exports.PaylaterTypeFromJSON = PaylaterTypeFromJSON;
function PaylaterTypeFromJSONTyped(json, ignoreDiscriminator) {
    if (json !== "" || json !== null) {
        const key = Object.keys(exports.PaylaterType)[Object.values(exports.PaylaterType).indexOf(json)];
        return exports.PaylaterType[key] === undefined ?
            exports.PaylaterType['XenditEnumDefaultFallback'] : exports.PaylaterType[key];
    }
    return json;
}
exports.PaylaterTypeFromJSONTyped = PaylaterTypeFromJSONTyped;
function PaylaterTypeToJSON(value) {
    return value;
}
exports.PaylaterTypeToJSON = PaylaterTypeToJSON;
