// xendit-node/invoice/models/QrCodeType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.QrCodeTypeToJSON = exports.QrCodeTypeFromJSONTyped = exports.QrCodeTypeFromJSON = exports.QrCodeType = void 0;
/**
 * Representing the available QR Code channels used for invoice-related transactions.
 * @export
 */
exports.QrCodeType = {
    Qris: 'QRIS',
    Promptpay: 'PROMPTPAY',
    XenditEnumDefaultFallback: "UNKNOWN_ENUM_VALUE"
};
function QrCodeTypeFromJSON(json) {
    return QrCodeTypeFromJSONTyped(json, false);
}
exports.QrCodeTypeFromJSON = QrCodeTypeFromJSON;
function QrCodeTypeFromJSONTyped(json, ignoreDiscriminator) {
    if (json !== "" || json !== null) {
        const key = Object.keys(exports.QrCodeType)[Object.values(exports.QrCodeType).indexOf(json)];
        return exports.QrCodeType[key] === undefined ?
            exports.QrCodeType['XenditEnumDefaultFallback'] : exports.QrCodeType[key];
    }
    return json;
}
exports.QrCodeTypeFromJSONTyped = QrCodeTypeFromJSONTyped;
function QrCodeTypeToJSON(value) {
    return value;
}
exports.QrCodeTypeToJSON = QrCodeTypeToJSON;
