// xendit-node/invoice/models/ServerError.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerErrorToJSON = exports.ServerErrorFromJSONTyped = exports.ServerErrorFromJSON = exports.instanceOfServerError = exports.ServerErrorErrorCodeEnum = void 0;
/**
 * @export
 */
exports.ServerErrorErrorCodeEnum = {
    ServerError: 'SERVER_ERROR'
};
/**
 * Check if a given object implements the ServerError interface.
 */
function instanceOfServerError(value) {
    let isInstance = true;
    isInstance = isInstance && "errorCode" in value;
    isInstance = isInstance && "message" in value;
    return isInstance;
}
exports.instanceOfServerError = instanceOfServerError;
function ServerErrorFromJSON(json) {
    return ServerErrorFromJSONTyped(json, false);
}
exports.ServerErrorFromJSON = ServerErrorFromJSON;
function ServerErrorFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'errorCode': json['error_code'],
        'message': json['message'],
    };
}
exports.ServerErrorFromJSONTyped = ServerErrorFromJSONTyped;
function ServerErrorToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'error_code': value.errorCode,
        'message': value.message,
    };
}
exports.ServerErrorToJSON = ServerErrorToJSON;
