// xendit-node/payment_method/models/CreatePaymentMethod503Response.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreatePaymentMethod503ResponseToJSON = exports.CreatePaymentMethod503ResponseFromJSONTyped = exports.CreatePaymentMethod503ResponseFromJSON = exports.instanceOfCreatePaymentMethod503Response = exports.CreatePaymentMethod503ResponseErrorCodeEnum = void 0;
const runtime_1 = require("../../runtime");
/**
 * @export
 */
exports.CreatePaymentMethod503ResponseErrorCodeEnum = {
    ChannelUnavailable: 'CHANNEL_UNAVAILABLE',
    OtpDeliveryError: 'OTP_DELIVERY_ERROR'
};
/**
 * Check if a given object implements the CreatePaymentMethod503Response interface.
 */
function instanceOfCreatePaymentMethod503Response(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfCreatePaymentMethod503Response = instanceOfCreatePaymentMethod503Response;
function CreatePaymentMethod503ResponseFromJSON(json) {
    return CreatePaymentMethod503ResponseFromJSONTyped(json, false);
}
exports.CreatePaymentMethod503ResponseFromJSON = CreatePaymentMethod503ResponseFromJSON;
function CreatePaymentMethod503ResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'errorCode': !(0, runtime_1.exists)(json, 'error_code') ? undefined : json['error_code'],
        'message': !(0, runtime_1.exists)(json, 'message') ? undefined : json['message'],
    };
}
exports.CreatePaymentMethod503ResponseFromJSONTyped = CreatePaymentMethod503ResponseFromJSONTyped;
function CreatePaymentMethod503ResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'error_code': value.errorCode,
        'message': value.message,
    };
}
exports.CreatePaymentMethod503ResponseToJSON = CreatePaymentMethod503ResponseToJSON;
