// xendit-node/payment_method/models/CreatePaymentMethod409Response.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreatePaymentMethod409ResponseToJSON = exports.CreatePaymentMethod409ResponseFromJSONTyped = exports.CreatePaymentMethod409ResponseFromJSON = exports.instanceOfCreatePaymentMethod409Response = exports.CreatePaymentMethod409ResponseErrorCodeEnum = void 0;
const runtime_1 = require("../../runtime");
/**
 * @export
 */
exports.CreatePaymentMethod409ResponseErrorCodeEnum = {
    PaymentMethodAlreadyActive: 'PAYMENT_METHOD_ALREADY_ACTIVE',
    PaymentMethodAlreadyFailed: 'PAYMENT_METHOD_ALREADY_FAILED',
    DuplicateError: 'DUPLICATE_ERROR',
    IdempotencyError: 'IDEMPOTENCY_ERROR'
};
/**
 * Check if a given object implements the CreatePaymentMethod409Response interface.
 */
function instanceOfCreatePaymentMethod409Response(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfCreatePaymentMethod409Response = instanceOfCreatePaymentMethod409Response;
function CreatePaymentMethod409ResponseFromJSON(json) {
    return CreatePaymentMethod409ResponseFromJSONTyped(json, false);
}
exports.CreatePaymentMethod409ResponseFromJSON = CreatePaymentMethod409ResponseFromJSON;
function CreatePaymentMethod409ResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'errorCode': !(0, runtime_1.exists)(json, 'error_code') ? undefined : json['error_code'],
        'message': !(0, runtime_1.exists)(json, 'message') ? undefined : json['message'],
    };
}
exports.CreatePaymentMethod409ResponseFromJSONTyped = CreatePaymentMethod409ResponseFromJSONTyped;
function CreatePaymentMethod409ResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'error_code': value.errorCode,
        'message': value.message,
    };
}
exports.CreatePaymentMethod409ResponseToJSON = CreatePaymentMethod409ResponseToJSON;
