// xendit-node/payment_method/models/DirectDebitType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DirectDebitTypeToJSON = exports.DirectDebitTypeFromJSONTyped = exports.DirectDebitTypeFromJSON = exports.DirectDebitType = void 0;
/**
 *
 * @export
 */
exports.DirectDebitType = {
    DebitCard: 'DEBIT_CARD',
    BankAccount: 'BANK_ACCOUNT',
    BankRedirect: 'BANK_REDIRECT',
    XenditEnumDefaultFallback: "UNKNOWN_ENUM_VALUE"
};
function DirectDebitTypeFromJSON(json) {
    return DirectDebitTypeFromJSONTyped(json, false);
}
exports.DirectDebitTypeFromJSON = DirectDebitTypeFromJSON;
function DirectDebitTypeFromJSONTyped(json, ignoreDiscriminator) {
    if (json !== "" || json !== null) {
        const key = Object.keys(exports.DirectDebitType)[Object.values(exports.DirectDebitType).indexOf(json)];
        return exports.DirectDebitType[key] === undefined ?
            exports.DirectDebitType['XenditEnumDefaultFallback'] : exports.DirectDebitType[key];
    }
    return json;
}
exports.DirectDebitTypeFromJSONTyped = DirectDebitTypeFromJSONTyped;
function DirectDebitTypeToJSON(value) {
    return value;
}
exports.DirectDebitTypeToJSON = DirectDebitTypeToJSON;
