// xendit-node/payment_method/models/GetAllPaymentMethods403Response.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetAllPaymentMethods403ResponseToJSON = exports.GetAllPaymentMethods403ResponseFromJSONTyped = exports.GetAllPaymentMethods403ResponseFromJSON = exports.instanceOfGetAllPaymentMethods403Response = exports.GetAllPaymentMethods403ResponseErrorCodeEnum = void 0;
const runtime_1 = require("../../runtime");
/**
 * @export
 */
exports.GetAllPaymentMethods403ResponseErrorCodeEnum = {
    Unauthorized: 'UNAUTHORIZED',
    ChannelNotActivated: 'CHANNEL_NOT_ACTIVATED'
};
/**
 * Check if a given object implements the GetAllPaymentMethods403Response interface.
 */
function instanceOfGetAllPaymentMethods403Response(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfGetAllPaymentMethods403Response = instanceOfGetAllPaymentMethods403Response;
function GetAllPaymentMethods403ResponseFromJSON(json) {
    return GetAllPaymentMethods403ResponseFromJSONTyped(json, false);
}
exports.GetAllPaymentMethods403ResponseFromJSON = GetAllPaymentMethods403ResponseFromJSON;
function GetAllPaymentMethods403ResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'errorCode': !(0, runtime_1.exists)(json, 'error_code') ? undefined : json['error_code'],
        'message': !(0, runtime_1.exists)(json, 'message') ? undefined : json['message'],
    };
}
exports.GetAllPaymentMethods403ResponseFromJSONTyped = GetAllPaymentMethods403ResponseFromJSONTyped;
function GetAllPaymentMethods403ResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'error_code': value.errorCode,
        'message': value.message,
    };
}
exports.GetAllPaymentMethods403ResponseToJSON = GetAllPaymentMethods403ResponseToJSON;
