// xendit-node/payment_method/models/GetAllPaymentMethods400Response.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetAllPaymentMethods400ResponseToJSON = exports.GetAllPaymentMethods400ResponseFromJSONTyped = exports.GetAllPaymentMethods400ResponseFromJSON = exports.instanceOfGetAllPaymentMethods400Response = exports.GetAllPaymentMethods400ResponseErrorCodeEnum = void 0;
const runtime_1 = require("../../runtime");
/**
 * @export
 */
exports.GetAllPaymentMethods400ResponseErrorCodeEnum = {
    ApiValidationError: 'API_VALIDATION_ERROR',
    NotSupportedByChannel: 'NOT_SUPPORTED_BY_CHANNEL',
    ExpiredOtpError: 'EXPIRED_OTP_ERROR',
    InvalidAccountDetails: 'INVALID_ACCOUNT_DETAILS',
    InvalidOtpError: 'INVALID_OTP_ERROR',
    InvalidPaymentMethod: 'INVALID_PAYMENT_METHOD',
    MaxOtpAttemptsError: 'MAX_OTP_ATTEMPTS_ERROR',
    CustomerNotFoundError: 'CUSTOMER_NOT_FOUND_ERROR',
    AccountAccessBlocked: 'ACCOUNT_ACCESS_BLOCKED',
    MaxAccountLinking: 'MAX_ACCOUNT_LINKING',
    InactivePaymentMethod: 'INACTIVE_PAYMENT_METHOD',
    PaymentMethodNotSupported: 'PAYMENT_METHOD_NOT_SUPPORTED',
    IncorrectAmount: 'INCORRECT_AMOUNT',
    DuplicatedFixedPaymentInstrument: 'DUPLICATED_FIXED_PAYMENT_INSTRUMENT',
    CurrencyMismatched: 'CURRENCY_MISMATCHED'
};
/**
 * Check if a given object implements the GetAllPaymentMethods400Response interface.
 */
function instanceOfGetAllPaymentMethods400Response(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfGetAllPaymentMethods400Response = instanceOfGetAllPaymentMethods400Response;
function GetAllPaymentMethods400ResponseFromJSON(json) {
    return GetAllPaymentMethods400ResponseFromJSONTyped(json, false);
}
exports.GetAllPaymentMethods400ResponseFromJSON = GetAllPaymentMethods400ResponseFromJSON;
function GetAllPaymentMethods400ResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'errorCode': !(0, runtime_1.exists)(json, 'error_code') ? undefined : json['error_code'],
        'message': !(0, runtime_1.exists)(json, 'message') ? undefined : json['message'],
    };
}
exports.GetAllPaymentMethods400ResponseFromJSONTyped = GetAllPaymentMethods400ResponseFromJSONTyped;
function GetAllPaymentMethods400ResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'error_code': value.errorCode,
        'message': value.message,
    };
}
exports.GetAllPaymentMethods400ResponseToJSON = GetAllPaymentMethods400ResponseToJSON;
