// xendit-node/payment_method/models/PaymentMethodCountry.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentMethodCountryToJSON = exports.PaymentMethodCountryFromJSONTyped = exports.PaymentMethodCountryFromJSON = exports.PaymentMethodCountry = void 0;
/**
 *
 * @export
 */
exports.PaymentMethodCountry = {
    Ph: 'PH',
    Id: 'ID',
    Vn: 'VN',
    Th: 'TH',
    My: 'MY',
    Us: 'US',
    XenditEnumDefaultFallback: "UNKNOWN_ENUM_VALUE"
};
function PaymentMethodCountryFromJSON(json) {
    return PaymentMethodCountryFromJSONTyped(json, false);
}
exports.PaymentMethodCountryFromJSON = PaymentMethodCountryFromJSON;
function PaymentMethodCountryFromJSONTyped(json, ignoreDiscriminator) {
    if (json !== "" || json !== null) {
        const key = Object.keys(exports.PaymentMethodCountry)[Object.values(exports.PaymentMethodCountry).indexOf(json)];
        return exports.PaymentMethodCountry[key] === undefined ?
            exports.PaymentMethodCountry['XenditEnumDefaultFallback'] : exports.PaymentMethodCountry[key];
    }
    return json;
}
exports.PaymentMethodCountryFromJSONTyped = PaymentMethodCountryFromJSONTyped;
function PaymentMethodCountryToJSON(value) {
    return value;
}
exports.PaymentMethodCountryToJSON = PaymentMethodCountryToJSON;
