// xendit-node/payment_method/models/PaymentMethodType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentMethodTypeToJSON = exports.PaymentMethodTypeFromJSONTyped = exports.PaymentMethodTypeFromJSON = exports.PaymentMethodType = void 0;
/**
 *
 * @export
 */
exports.PaymentMethodType = {
    Card: 'CARD',
    Cryptocurrency: 'CRYPTOCURRENCY',
    DirectBankTransfer: 'DIRECT_BANK_TRANSFER',
    DirectDebit: 'DIRECT_DEBIT',
    Ewallet: 'EWALLET',
    OverTheCounter: 'OVER_THE_COUNTER',
    QrCode: 'QR_CODE',
    VirtualAccount: 'VIRTUAL_ACCOUNT',
    XenditEnumDefaultFallback: "UNKNOWN_ENUM_VALUE"
};
function PaymentMethodTypeFromJSON(json) {
    return PaymentMethodTypeFromJSONTyped(json, false);
}
exports.PaymentMethodTypeFromJSON = PaymentMethodTypeFromJSON;
function PaymentMethodTypeFromJSONTyped(json, ignoreDiscriminator) {
    if (json !== "" || json !== null) {
        const key = Object.keys(exports.PaymentMethodType)[Object.values(exports.PaymentMethodType).indexOf(json)];
        return exports.PaymentMethodType[key] === undefined ?
            exports.PaymentMethodType['XenditEnumDefaultFallback'] : exports.PaymentMethodType[key];
    }
    return json;
}
exports.PaymentMethodTypeFromJSONTyped = PaymentMethodTypeFromJSONTyped;
function PaymentMethodTypeToJSON(value) {
    return value;
}
exports.PaymentMethodTypeToJSON = PaymentMethodTypeToJSON;
