// xendit-node/payment_method/models/PaymentMethodReusability.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentMethodReusabilityToJSON = exports.PaymentMethodReusabilityFromJSONTyped = exports.PaymentMethodReusabilityFromJSON = exports.PaymentMethodReusability = void 0;
/**
 *
 * @export
 */
exports.PaymentMethodReusability = {
    MultipleUse: 'MULTIPLE_USE',
    OneTimeUse: 'ONE_TIME_USE',
    XenditEnumDefaultFallback: "UNKNOWN_ENUM_VALUE"
};
function PaymentMethodReusabilityFromJSON(json) {
    return PaymentMethodReusabilityFromJSONTyped(json, false);
}
exports.PaymentMethodReusabilityFromJSON = PaymentMethodReusabilityFromJSON;
function PaymentMethodReusabilityFromJSONTyped(json, ignoreDiscriminator) {
    if (json !== "" || json !== null) {
        const key = Object.keys(exports.PaymentMethodReusability)[Object.values(exports.PaymentMethodReusability).indexOf(json)];
        return exports.PaymentMethodReusability[key] === undefined ?
            exports.PaymentMethodReusability['XenditEnumDefaultFallback'] : exports.PaymentMethodReusability[key];
    }
    return json;
}
exports.PaymentMethodReusabilityFromJSONTyped = PaymentMethodReusabilityFromJSONTyped;
function PaymentMethodReusabilityToJSON(value) {
    return value;
}
exports.PaymentMethodReusabilityToJSON = PaymentMethodReusabilityToJSON;
