// xendit-node/payment_method/models/VirtualAccountAlternativeDisplay.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualAccountAlternativeDisplayToJSON = exports.VirtualAccountAlternativeDisplayFromJSONTyped = exports.VirtualAccountAlternativeDisplayFromJSON = exports.instanceOfVirtualAccountAlternativeDisplay = exports.VirtualAccountAlternativeDisplayTypeEnum = void 0;
const runtime_1 = require("../../runtime");
/**
 * @export
 */
exports.VirtualAccountAlternativeDisplayTypeEnum = {
    QrString: 'QR_STRING'
};
/**
 * Check if a given object implements the VirtualAccountAlternativeDisplay interface.
 */
function instanceOfVirtualAccountAlternativeDisplay(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfVirtualAccountAlternativeDisplay = instanceOfVirtualAccountAlternativeDisplay;
function VirtualAccountAlternativeDisplayFromJSON(json) {
    return VirtualAccountAlternativeDisplayFromJSONTyped(json, false);
}
exports.VirtualAccountAlternativeDisplayFromJSON = VirtualAccountAlternativeDisplayFromJSON;
function VirtualAccountAlternativeDisplayFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': !(0, runtime_1.exists)(json, 'type') ? undefined : json['type'],
        'data': !(0, runtime_1.exists)(json, 'data') ? undefined : json['data'],
    };
}
exports.VirtualAccountAlternativeDisplayFromJSONTyped = VirtualAccountAlternativeDisplayFromJSONTyped;
function VirtualAccountAlternativeDisplayToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'data': value.data,
    };
}
exports.VirtualAccountAlternativeDisplayToJSON = VirtualAccountAlternativeDisplayToJSON;
