// xendit-node/payment_method/models/VirtualAccountChannelCode.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualAccountChannelCodeToJSON = exports.VirtualAccountChannelCodeFromJSONTyped = exports.VirtualAccountChannelCodeFromJSON = exports.VirtualAccountChannelCode = void 0;
/**
 * Virtual Account Channel Code
 * @export
 */
exports.VirtualAccountChannelCode = {
    Bca: 'BCA',
    Bjb: 'BJB',
    Bni: 'BNI',
    Bri: 'BRI',
    Mandiri: 'MANDIRI',
    Permata: 'PERMATA',
    Bsi: 'BSI',
    Cimb: 'CIMB',
    SahabatSampoerna: 'SAHABAT_SAMPOERNA',
    Artajasa: 'ARTAJASA',
    Pv: 'PV',
    Vietcapital: 'VIETCAPITAL',
    Woori: 'WOORI',
    Msb: 'MSB',
    StandardChartered: 'STANDARD_CHARTERED',
    Ambank: 'AMBANK',
    Bnc: 'BNC',
    Hana: 'HANA',
    Muamalat: 'MUAMALAT',
    XenditEnumDefaultFallback: "UNKNOWN_ENUM_VALUE"
};
function VirtualAccountChannelCodeFromJSON(json) {
    return VirtualAccountChannelCodeFromJSONTyped(json, false);
}
exports.VirtualAccountChannelCodeFromJSON = VirtualAccountChannelCodeFromJSON;
function VirtualAccountChannelCodeFromJSONTyped(json, ignoreDiscriminator) {
    if (json !== "" || json !== null) {
        const key = Object.keys(exports.VirtualAccountChannelCode)[Object.values(exports.VirtualAccountChannelCode).indexOf(json)];
        return exports.VirtualAccountChannelCode[key] === undefined ?
            exports.VirtualAccountChannelCode['XenditEnumDefaultFallback'] : exports.VirtualAccountChannelCode[key];
    }
    return json;
}
exports.VirtualAccountChannelCodeFromJSONTyped = VirtualAccountChannelCodeFromJSONTyped;
function VirtualAccountChannelCodeToJSON(value) {
    return value;
}
exports.VirtualAccountChannelCodeToJSON = VirtualAccountChannelCodeToJSON;
