// xendit-node/payment_method/models/VirtualAccountUpdateParameters.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualAccountUpdateParametersToJSON = exports.VirtualAccountUpdateParametersFromJSONTyped = exports.VirtualAccountUpdateParametersFromJSON = exports.instanceOfVirtualAccountUpdateParameters = exports.VirtualAccountUpdateParametersAlternativeDisplayTypesEnum = void 0;
const runtime_1 = require("../../runtime");
const VirtualAccountChannelPropertiesPatch_1 = require("./VirtualAccountChannelPropertiesPatch");
/**
 * @export
 */
exports.VirtualAccountUpdateParametersAlternativeDisplayTypesEnum = {
    QrString: 'QR_STRING'
};
/**
 * Check if a given object implements the VirtualAccountUpdateParameters interface.
 */
function instanceOfVirtualAccountUpdateParameters(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfVirtualAccountUpdateParameters = instanceOfVirtualAccountUpdateParameters;
function VirtualAccountUpdateParametersFromJSON(json) {
    return VirtualAccountUpdateParametersFromJSONTyped(json, false);
}
exports.VirtualAccountUpdateParametersFromJSON = VirtualAccountUpdateParametersFromJSON;
function VirtualAccountUpdateParametersFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'amount': !(0, runtime_1.exists)(json, 'amount') ? undefined : json['amount'],
        'minAmount': !(0, runtime_1.exists)(json, 'min_amount') ? undefined : json['min_amount'],
        'maxAmount': !(0, runtime_1.exists)(json, 'max_amount') ? undefined : json['max_amount'],
        'channelProperties': !(0, runtime_1.exists)(json, 'channel_properties') ? undefined : (0, VirtualAccountChannelPropertiesPatch_1.VirtualAccountChannelPropertiesPatchFromJSON)(json['channel_properties']),
        'alternativeDisplayTypes': !(0, runtime_1.exists)(json, 'alternative_display_types') ? undefined : json['alternative_display_types'],
    };
}
exports.VirtualAccountUpdateParametersFromJSONTyped = VirtualAccountUpdateParametersFromJSONTyped;
function VirtualAccountUpdateParametersToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'amount': value.amount,
        'min_amount': value.minAmount,
        'max_amount': value.maxAmount,
        'channel_properties': (0, VirtualAccountChannelPropertiesPatch_1.VirtualAccountChannelPropertiesPatchToJSON)(value.channelProperties),
        'alternative_display_types': value.alternativeDisplayTypes,
    };
}
exports.VirtualAccountUpdateParametersToJSON = VirtualAccountUpdateParametersToJSON;
