// xendit-node/payment_request/models/CardInformation.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardInformationToJSON = exports.CardInformationFromJSONTyped = exports.CardInformationFromJSON = exports.instanceOfCardInformation = exports.CardInformationNetworkEnum = void 0;
const runtime_1 = require("../../runtime");
/**
 * @export
 */
exports.CardInformationNetworkEnum = {
    Visa: 'VISA',
    Mastercard: 'MASTERCARD',
    Jcb: 'JCB',
    Amex: 'AMEX',
    Discover: 'DISCOVER'
};
/**
 * Check if a given object implements the CardInformation interface.
 */
function instanceOfCardInformation(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfCardInformation = instanceOfCardInformation;
function CardInformationFromJSON(json) {
    return CardInformationFromJSONTyped(json, false);
}
exports.CardInformationFromJSON = CardInformationFromJSON;
function CardInformationFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'tokenId': !(0, runtime_1.exists)(json, 'token_id') ? undefined : json['token_id'],
        'maskedCardNumber': !(0, runtime_1.exists)(json, 'masked_card_number') ? undefined : json['masked_card_number'],
        'expiryMonth': !(0, runtime_1.exists)(json, 'expiry_month') ? undefined : json['expiry_month'],
        'expiryYear': !(0, runtime_1.exists)(json, 'expiry_year') ? undefined : json['expiry_year'],
        'cardholderName': !(0, runtime_1.exists)(json, 'cardholder_name') ? undefined : json['cardholder_name'],
        'fingerprint': !(0, runtime_1.exists)(json, 'fingerprint') ? undefined : json['fingerprint'],
        'type': !(0, runtime_1.exists)(json, 'type') ? undefined : json['type'],
        'network': !(0, runtime_1.exists)(json, 'network') ? undefined : json['network'],
        'country': !(0, runtime_1.exists)(json, 'country') ? undefined : json['country'],
        'issuer': !(0, runtime_1.exists)(json, 'issuer') ? undefined : json['issuer'],
        'cardNumber': !(0, runtime_1.exists)(json, 'card_number') ? undefined : json['card_number'],
        'oneTimeToken': !(0, runtime_1.exists)(json, 'one_time_token') ? undefined : json['one_time_token'],
    };
}
exports.CardInformationFromJSONTyped = CardInformationFromJSONTyped;
function CardInformationToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'token_id': value.tokenId,
        'masked_card_number': value.maskedCardNumber,
        'expiry_month': value.expiryMonth,
        'expiry_year': value.expiryYear,
        'cardholder_name': value.cardholderName,
        'fingerprint': value.fingerprint,
        'type': value.type,
        'network': value.network,
        'country': value.country,
        'issuer': value.issuer,
        'card_number': value.cardNumber,
        'one_time_token': value.oneTimeToken,
    };
}
exports.CardInformationToJSON = CardInformationToJSON;
