// xendit-node/payment_request/models/CardVerificationResultsThreeDSecure.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardVerificationResultsThreeDSecureToJSON = exports.CardVerificationResultsThreeDSecureFromJSONTyped = exports.CardVerificationResultsThreeDSecureFromJSON = exports.instanceOfCardVerificationResultsThreeDSecure = exports.CardVerificationResultsThreeDSecureThreeDSecureFlowEnum = void 0;
const runtime_1 = require("../../runtime");
/**
 * @export
 */
exports.CardVerificationResultsThreeDSecureThreeDSecureFlowEnum = {
    Challenge: 'CHALLENGE',
    Frictionless: 'FRICTIONLESS'
};
/**
 * Check if a given object implements the CardVerificationResultsThreeDSecure interface.
 */
function instanceOfCardVerificationResultsThreeDSecure(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfCardVerificationResultsThreeDSecure = instanceOfCardVerificationResultsThreeDSecure;
function CardVerificationResultsThreeDSecureFromJSON(json) {
    return CardVerificationResultsThreeDSecureFromJSONTyped(json, false);
}
exports.CardVerificationResultsThreeDSecureFromJSON = CardVerificationResultsThreeDSecureFromJSON;
function CardVerificationResultsThreeDSecureFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'threeDSecureFlow': !(0, runtime_1.exists)(json, 'three_d_secure_flow') ? undefined : json['three_d_secure_flow'],
        'eciCode': !(0, runtime_1.exists)(json, 'eci_code') ? undefined : json['eci_code'],
        'threeDSecureResult': !(0, runtime_1.exists)(json, 'three_d_secure_result') ? undefined : json['three_d_secure_result'],
        'threeDSecureResultReason': !(0, runtime_1.exists)(json, 'three_d_secure_result_reason') ? undefined : json['three_d_secure_result_reason'],
        'threeDSecureVersion': !(0, runtime_1.exists)(json, 'three_d_secure_version') ? undefined : json['three_d_secure_version'],
    };
}
exports.CardVerificationResultsThreeDSecureFromJSONTyped = CardVerificationResultsThreeDSecureFromJSONTyped;
function CardVerificationResultsThreeDSecureToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'three_d_secure_flow': value.threeDSecureFlow,
        'eci_code': value.eciCode,
        'three_d_secure_result': value.threeDSecureResult,
        'three_d_secure_result_reason': value.threeDSecureResultReason,
        'three_d_secure_version': value.threeDSecureVersion,
    };
}
exports.CardVerificationResultsThreeDSecureToJSON = CardVerificationResultsThreeDSecureToJSON;
