// xendit-node/payment_request/models/CardChannelCode.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardChannelCodeToJSON = exports.CardChannelCodeFromJSONTyped = exports.CardChannelCodeFromJSON = exports.CardChannelCode = void 0;
/**
 * Card Channel Code
 * @export
 */
exports.CardChannelCode = {
    Gpn: 'GPN',
    XenditEnumDefaultFallback: "UNKNOWN_ENUM_VALUE"
};
function CardChannelCodeFromJSON(json) {
    return CardChannelCodeFromJSONTyped(json, false);
}
exports.CardChannelCodeFromJSON = CardChannelCodeFromJSON;
function CardChannelCodeFromJSONTyped(json, ignoreDiscriminator) {
    if (json !== "" || json !== null) {
        const key = Object.keys(exports.CardChannelCode)[Object.values(exports.CardChannelCode).indexOf(json)];
        return exports.CardChannelCode[key] === undefined ?
            exports.CardChannelCode['XenditEnumDefaultFallback'] : exports.CardChannelCode[key];
    }
    return json;
}
exports.CardChannelCodeFromJSONTyped = CardChannelCodeFromJSONTyped;
function CardChannelCodeToJSON(value) {
    return value;
}
exports.CardChannelCodeToJSON = CardChannelCodeToJSON;
