// xendit-node/payment_request/models/EWalletChannelCode.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EWalletChannelCodeToJSON = exports.EWalletChannelCodeFromJSONTyped = exports.EWalletChannelCodeFromJSON = exports.EWalletChannelCode = void 0;
/**
 * Ewallet Channel Code
 * @export
 */
exports.EWalletChannelCode = {
    Gcash: 'GCASH',
    Grabpay: 'GRABPAY',
    Paymaya: 'PAYMAYA',
    Dana: 'DANA',
    Ovo: 'OVO',
    Linkaja: 'LINKAJA',
    Shopeepay: 'SHOPEEPAY',
    Nexcash: 'NEXCASH',
    Astrapay: 'ASTRAPAY',
    Jeniuspay: 'JENIUSPAY',
    Appota: 'APPOTA',
    Momo: 'MOMO',
    Vnptwallet: 'VNPTWALLET',
    Viettelpay: 'VIETTELPAY',
    Zalopay: 'ZALOPAY',
    Wechatpay: 'WECHATPAY',
    Linepay: 'LINEPAY',
    Truemoney: 'TRUEMONEY',
    Alipay: 'ALIPAY',
    Touchngo: 'TOUCHNGO',
    XenditEnumDefaultFallback: "UNKNOWN_ENUM_VALUE"
};
function EWalletChannelCodeFromJSON(json) {
    return EWalletChannelCodeFromJSONTyped(json, false);
}
exports.EWalletChannelCodeFromJSON = EWalletChannelCodeFromJSON;
function EWalletChannelCodeFromJSONTyped(json, ignoreDiscriminator) {
    if (json !== "" || json !== null) {
        const key = Object.keys(exports.EWalletChannelCode)[Object.values(exports.EWalletChannelCode).indexOf(json)];
        return exports.EWalletChannelCode[key] === undefined ?
            exports.EWalletChannelCode['XenditEnumDefaultFallback'] : exports.EWalletChannelCode[key];
    }
    return json;
}
exports.EWalletChannelCodeFromJSONTyped = EWalletChannelCodeFromJSONTyped;
function EWalletChannelCodeToJSON(value) {
    return value;
}
exports.EWalletChannelCodeToJSON = EWalletChannelCodeToJSON;
