// xendit-node/payment_request/models/PaymentRequestInitiator.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentRequestInitiatorToJSON = exports.PaymentRequestInitiatorFromJSONTyped = exports.PaymentRequestInitiatorFromJSON = exports.PaymentRequestInitiator = void 0;
/**
 *
 * @export
 */
exports.PaymentRequestInitiator = {
    Customer: 'CUSTOMER',
    Merchant: 'MERCHANT',
    XenditEnumDefaultFallback: "UNKNOWN_ENUM_VALUE"
};
function PaymentRequestInitiatorFromJSON(json) {
    return PaymentRequestInitiatorFromJSONTyped(json, false);
}
exports.PaymentRequestInitiatorFromJSON = PaymentRequestInitiatorFromJSON;
function PaymentRequestInitiatorFromJSONTyped(json, ignoreDiscriminator) {
    if (json !== "" || json !== null) {
        const key = Object.keys(exports.PaymentRequestInitiator)[Object.values(exports.PaymentRequestInitiator).indexOf(json)];
        return exports.PaymentRequestInitiator[key] === undefined ?
            exports.PaymentRequestInitiator['XenditEnumDefaultFallback'] : exports.PaymentRequestInitiator[key];
    }
    return json;
}
exports.PaymentRequestInitiatorFromJSONTyped = PaymentRequestInitiatorFromJSONTyped;
function PaymentRequestInitiatorToJSON(value) {
    return value;
}
exports.PaymentRequestInitiatorToJSON = PaymentRequestInitiatorToJSON;
