// xendit-node/payment_request/models/PaymentRequestCurrency.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentRequestCurrencyToJSON = exports.PaymentRequestCurrencyFromJSONTyped = exports.PaymentRequestCurrencyFromJSON = exports.PaymentRequestCurrency = void 0;
/**
 *
 * @export
 */
exports.PaymentRequestCurrency = {
    Idr: 'IDR',
    Php: 'PHP',
    Vnd: 'VND',
    Thb: 'THB',
    Myr: 'MYR',
    Usd: 'USD',
    XenditEnumDefaultFallback: "UNKNOWN_ENUM_VALUE"
};
function PaymentRequestCurrencyFromJSON(json) {
    return PaymentRequestCurrencyFromJSONTyped(json, false);
}
exports.PaymentRequestCurrencyFromJSON = PaymentRequestCurrencyFromJSON;
function PaymentRequestCurrencyFromJSONTyped(json, ignoreDiscriminator) {
    if (json !== "" || json !== null) {
        const key = Object.keys(exports.PaymentRequestCurrency)[Object.values(exports.PaymentRequestCurrency).indexOf(json)];
        return exports.PaymentRequestCurrency[key] === undefined ?
            exports.PaymentRequestCurrency['XenditEnumDefaultFallback'] : exports.PaymentRequestCurrency[key];
    }
    return json;
}
exports.PaymentRequestCurrencyFromJSONTyped = PaymentRequestCurrencyFromJSONTyped;
function PaymentRequestCurrencyToJSON(value) {
    return value;
}
exports.PaymentRequestCurrencyToJSON = PaymentRequestCurrencyToJSON;
