// xendit-node/payment_request/models/PaymentMethodStatus.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentMethodStatusToJSON = exports.PaymentMethodStatusFromJSONTyped = exports.PaymentMethodStatusFromJSON = exports.PaymentMethodStatus = void 0;
/**
 *
 * @export
 */
exports.PaymentMethodStatus = {
    Active: 'ACTIVE',
    Inactive: 'INACTIVE',
    Pending: 'PENDING',
    Expired: 'EXPIRED',
    Failed: 'FAILED',
    XenditEnumDefaultFallback: "UNKNOWN_ENUM_VALUE"
};
function PaymentMethodStatusFromJSON(json) {
    return PaymentMethodStatusFromJSONTyped(json, false);
}
exports.PaymentMethodStatusFromJSON = PaymentMethodStatusFromJSON;
function PaymentMethodStatusFromJSONTyped(json, ignoreDiscriminator) {
    if (json !== "" || json !== null) {
        const key = Object.keys(exports.PaymentMethodStatus)[Object.values(exports.PaymentMethodStatus).indexOf(json)];
        return exports.PaymentMethodStatus[key] === undefined ?
            exports.PaymentMethodStatus['XenditEnumDefaultFallback'] : exports.PaymentMethodStatus[key];
    }
    return json;
}
exports.PaymentMethodStatusFromJSONTyped = PaymentMethodStatusFromJSONTyped;
function PaymentMethodStatusToJSON(value) {
    return value;
}
exports.PaymentMethodStatusToJSON = PaymentMethodStatusToJSON;
