// xendit-node/payment_request/models/PaymentRequestStatus.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentRequestStatusToJSON = exports.PaymentRequestStatusFromJSONTyped = exports.PaymentRequestStatusFromJSON = exports.PaymentRequestStatus = void 0;
/**
 *
 * @export
 */
exports.PaymentRequestStatus = {
    Pending: 'PENDING',
    RequiresAction: 'REQUIRES_ACTION',
    Canceled: 'CANCELED',
    Succeeded: 'SUCCEEDED',
    Failed: 'FAILED',
    Voided: 'VOIDED',
    Unknown: 'UNKNOWN',
    AwaitingCapture: 'AWAITING_CAPTURE',
    Expired: 'EXPIRED',
    XenditEnumDefaultFallback: "UNKNOWN_ENUM_VALUE"
};
function PaymentRequestStatusFromJSON(json) {
    return PaymentRequestStatusFromJSONTyped(json, false);
}
exports.PaymentRequestStatusFromJSON = PaymentRequestStatusFromJSON;
function PaymentRequestStatusFromJSONTyped(json, ignoreDiscriminator) {
    if (json !== "" || json !== null) {
        const key = Object.keys(exports.PaymentRequestStatus)[Object.values(exports.PaymentRequestStatus).indexOf(json)];
        return exports.PaymentRequestStatus[key] === undefined ?
            exports.PaymentRequestStatus['XenditEnumDefaultFallback'] : exports.PaymentRequestStatus[key];
    }
    return json;
}
exports.PaymentRequestStatusFromJSONTyped = PaymentRequestStatusFromJSONTyped;
function PaymentRequestStatusToJSON(value) {
    return value;
}
exports.PaymentRequestStatusToJSON = PaymentRequestStatusToJSON;
