// xendit-node/balance_and_transaction/models/ChannelsCategories.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChannelsCategoriesToJSON = exports.ChannelsCategoriesFromJSONTyped = exports.ChannelsCategoriesFromJSON = exports.ChannelsCategories = void 0;
/**
 *
 * @export
 */
exports.ChannelsCategories = {
    Bank: 'BANK',
    CardlessCredit: 'CARDLESS_CREDIT',
    Paylater: 'PAYLATER',
    Cards: 'CARDS',
    Cash: 'CASH',
    DirectDebit: 'DIRECT_DEBIT',
    Ewallet: 'EWALLET',
    Invoice: 'INVOICE',
    QrCode: 'QR_CODE',
    RetailOutlet: 'RETAIL_OUTLET',
    VirtualAccount: 'VIRTUAL_ACCOUNT',
    Xenplatform: 'XENPLATFORM',
    DirectBankTransfer: 'DIRECT_BANK_TRANSFER',
    Other: 'OTHER',
    XenditEnumDefaultFallback: "UNKNOWN_ENUM_VALUE"
};
function ChannelsCategoriesFromJSON(json) {
    return ChannelsCategoriesFromJSONTyped(json, false);
}
exports.ChannelsCategoriesFromJSON = ChannelsCategoriesFromJSON;
function ChannelsCategoriesFromJSONTyped(json, ignoreDiscriminator) {
    if (json !== "" || json !== null) {
        const key = Object.keys(exports.ChannelsCategories)[Object.values(exports.ChannelsCategories).indexOf(json)];
        return exports.ChannelsCategories[key] === undefined ?
            exports.ChannelsCategories['XenditEnumDefaultFallback'] : exports.ChannelsCategories[key];
    }
    return json;
}
exports.ChannelsCategoriesFromJSONTyped = ChannelsCategoriesFromJSONTyped;
function ChannelsCategoriesToJSON(value) {
    return value;
}
exports.ChannelsCategoriesToJSON = ChannelsCategoriesToJSON;
