// xendit-node/customer/models/AddressStatus.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddressStatusToJSON = exports.AddressStatusFromJSONTyped = exports.AddressStatusFromJSON = exports.AddressStatus = void 0;
/**
 *
 * @export
 */
exports.AddressStatus = {
    Active: 'ACTIVE',
    Deleted: 'DELETED',
    XenditEnumDefaultFallback: "UNKNOWN_ENUM_VALUE"
};
function AddressStatusFromJSON(json) {
    return AddressStatusFromJSONTyped(json, false);
}
exports.AddressStatusFromJSON = AddressStatusFromJSON;
function AddressStatusFromJSONTyped(json, ignoreDiscriminator) {
    if (json !== "" || json !== null) {
        const key = Object.keys(exports.AddressStatus)[Object.values(exports.AddressStatus).indexOf(json)];
        return exports.AddressStatus[key] === undefined ?
            exports.AddressStatus['XenditEnumDefaultFallback'] : exports.AddressStatus[key];
    }
    return json;
}
exports.AddressStatusFromJSONTyped = AddressStatusFromJSONTyped;
function AddressStatusToJSON(value) {
    return value;
}
exports.AddressStatusToJSON = AddressStatusToJSON;
