// xendit-node/customer/models/BusinessDetail.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BusinessDetailToJSON = exports.BusinessDetailFromJSONTyped = exports.BusinessDetailFromJSON = exports.instanceOfBusinessDetail = exports.BusinessDetailBusinessTypeEnum = void 0;
const runtime_1 = require("../../runtime");
/**
 * @export
 */
exports.BusinessDetailBusinessTypeEnum = {
    Corporation: 'CORPORATION',
    SoleProprietor: 'SOLE_PROPRIETOR',
    Partnership: 'PARTNERSHIP',
    Cooperative: 'COOPERATIVE',
    Trust: 'TRUST',
    NonProfit: 'NON_PROFIT',
    Government: 'GOVERNMENT'
};
/**
 * Check if a given object implements the BusinessDetail interface.
 */
function instanceOfBusinessDetail(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfBusinessDetail = instanceOfBusinessDetail;
function BusinessDetailFromJSON(json) {
    return BusinessDetailFromJSONTyped(json, false);
}
exports.BusinessDetailFromJSON = BusinessDetailFromJSON;
function BusinessDetailFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'businessName': !(0, runtime_1.exists)(json, 'business_name') ? undefined : json['business_name'],
        'businessType': !(0, runtime_1.exists)(json, 'business_type') ? undefined : json['business_type'],
        'dateOfRegistration': !(0, runtime_1.exists)(json, 'date_of_registration') ? undefined : json['date_of_registration'],
        'natureOfBusiness': !(0, runtime_1.exists)(json, 'nature_of_business') ? undefined : json['nature_of_business'],
        'businessDomicile': !(0, runtime_1.exists)(json, 'business_domicile') ? undefined : json['business_domicile'],
    };
}
exports.BusinessDetailFromJSONTyped = BusinessDetailFromJSONTyped;
function BusinessDetailToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'business_name': value.businessName,
        'business_type': value.businessType,
        'date_of_registration': value.dateOfRegistration,
        'nature_of_business': value.natureOfBusiness,
        'business_domicile': value.businessDomicile,
    };
}
exports.BusinessDetailToJSON = BusinessDetailToJSON;
