// xendit-node/customer/models/CreateCustomer400Response.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateCustomer400ResponseToJSON = exports.CreateCustomer400ResponseFromJSONTyped = exports.CreateCustomer400ResponseFromJSON = exports.instanceOfCreateCustomer400Response = exports.CreateCustomer400ResponseErrorCodeEnum = void 0;
const runtime_1 = require("../../runtime");
/**
 * @export
 */
exports.CreateCustomer400ResponseErrorCodeEnum = {
    DuplicateEndCustomerError: 'DUPLICATE_END_CUSTOMER_ERROR',
    ApiValidationError: 'API_VALIDATION_ERROR'
};
/**
 * Check if a given object implements the CreateCustomer400Response interface.
 */
function instanceOfCreateCustomer400Response(value) {
    let isInstance = true;
    isInstance = isInstance && "errorCode" in value;
    isInstance = isInstance && "message" in value;
    return isInstance;
}
exports.instanceOfCreateCustomer400Response = instanceOfCreateCustomer400Response;
function CreateCustomer400ResponseFromJSON(json) {
    return CreateCustomer400ResponseFromJSONTyped(json, false);
}
exports.CreateCustomer400ResponseFromJSON = CreateCustomer400ResponseFromJSON;
function CreateCustomer400ResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'errorCode': json['error_code'],
        'message': json['message'],
        'errors': !(0, runtime_1.exists)(json, 'errors') ? undefined : json['errors'],
    };
}
exports.CreateCustomer400ResponseFromJSONTyped = CreateCustomer400ResponseFromJSONTyped;
function CreateCustomer400ResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'error_code': value.errorCode,
        'message': value.message,
        'errors': value.errors,
    };
}
exports.CreateCustomer400ResponseToJSON = CreateCustomer400ResponseToJSON;
