// xendit-node/customer/models/CustomerRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerRequestToJSON = exports.CustomerRequestFromJSONTyped = exports.CustomerRequestFromJSON = exports.instanceOfCustomerRequest = exports.CustomerRequestTypeEnum = void 0;
const runtime_1 = require("../../runtime");
const AddressRequest_1 = require("./AddressRequest");
const BusinessDetail_1 = require("./BusinessDetail");
const IdentityAccountRequest_1 = require("./IdentityAccountRequest");
const IndividualDetail_1 = require("./IndividualDetail");
const KYCDocumentRequest_1 = require("./KYCDocumentRequest");
/**
 * @export
 */
exports.CustomerRequestTypeEnum = {
    Individual: 'INDIVIDUAL',
    Business: 'BUSINESS'
};
/**
 * Check if a given object implements the CustomerRequest interface.
 */
function instanceOfCustomerRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "referenceId" in value;
    return isInstance;
}
exports.instanceOfCustomerRequest = instanceOfCustomerRequest;
function CustomerRequestFromJSON(json) {
    return CustomerRequestFromJSONTyped(json, false);
}
exports.CustomerRequestFromJSON = CustomerRequestFromJSON;
function CustomerRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'clientName': !(0, runtime_1.exists)(json, 'client_name') ? undefined : json['client_name'],
        'referenceId': json['reference_id'],
        'type': !(0, runtime_1.exists)(json, 'type') ? undefined : json['type'],
        'individualDetail': !(0, runtime_1.exists)(json, 'individual_detail') ? undefined : (0, IndividualDetail_1.IndividualDetailFromJSON)(json['individual_detail']),
        'businessDetail': !(0, runtime_1.exists)(json, 'business_detail') ? undefined : (0, BusinessDetail_1.BusinessDetailFromJSON)(json['business_detail']),
        'description': !(0, runtime_1.exists)(json, 'description') ? undefined : json['description'],
        'email': !(0, runtime_1.exists)(json, 'email') ? undefined : json['email'],
        'mobileNumber': !(0, runtime_1.exists)(json, 'mobile_number') ? undefined : json['mobile_number'],
        'phoneNumber': !(0, runtime_1.exists)(json, 'phone_number') ? undefined : json['phone_number'],
        'addresses': !(0, runtime_1.exists)(json, 'addresses') ? undefined : (json['addresses'].map(AddressRequest_1.AddressRequestFromJSON)),
        'identityAccounts': !(0, runtime_1.exists)(json, 'identity_accounts') ? undefined : (json['identity_accounts'].map(IdentityAccountRequest_1.IdentityAccountRequestFromJSON)),
        'kycDocuments': !(0, runtime_1.exists)(json, 'kyc_documents') ? undefined : (json['kyc_documents'].map(KYCDocumentRequest_1.KYCDocumentRequestFromJSON)),
        'metadata': !(0, runtime_1.exists)(json, 'metadata') ? undefined : json['metadata'],
    };
}
exports.CustomerRequestFromJSONTyped = CustomerRequestFromJSONTyped;
function CustomerRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'client_name': value.clientName,
        'reference_id': value.referenceId,
        'type': value.type,
        'individual_detail': (0, IndividualDetail_1.IndividualDetailToJSON)(value.individualDetail),
        'business_detail': (0, BusinessDetail_1.BusinessDetailToJSON)(value.businessDetail),
        'description': value.description,
        'email': value.email,
        'mobile_number': value.mobileNumber,
        'phone_number': value.phoneNumber,
        'addresses': value.addresses === undefined ? undefined : (value.addresses.map(AddressRequest_1.AddressRequestToJSON)),
        'identity_accounts': value.identityAccounts === undefined ? undefined : (value.identityAccounts.map(IdentityAccountRequest_1.IdentityAccountRequestToJSON)),
        'kyc_documents': value.kycDocuments === undefined ? undefined : (value.kycDocuments.map(KYCDocumentRequest_1.KYCDocumentRequestToJSON)),
        'metadata': value.metadata,
    };
}
exports.CustomerRequestToJSON = CustomerRequestToJSON;
