// xendit-node/customer/models/EndCustomerStatus.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EndCustomerStatusToJSON = exports.EndCustomerStatusFromJSONTyped = exports.EndCustomerStatusFromJSON = exports.EndCustomerStatus = void 0;
/**
 *
 * @export
 */
exports.EndCustomerStatus = {
    Active: 'ACTIVE',
    Inactive: 'INACTIVE',
    Pending: 'PENDING',
    Blocked: 'BLOCKED',
    Deleted: 'DELETED',
    XenditEnumDefaultFallback: "UNKNOWN_ENUM_VALUE"
};
function EndCustomerStatusFromJSON(json) {
    return EndCustomerStatusFromJSONTyped(json, false);
}
exports.EndCustomerStatusFromJSON = EndCustomerStatusFromJSON;
function EndCustomerStatusFromJSONTyped(json, ignoreDiscriminator) {
    if (json !== "" || json !== null) {
        const key = Object.keys(exports.EndCustomerStatus)[Object.values(exports.EndCustomerStatus).indexOf(json)];
        return exports.EndCustomerStatus[key] === undefined ?
            exports.EndCustomerStatus['XenditEnumDefaultFallback'] : exports.EndCustomerStatus[key];
    }
    return json;
}
exports.EndCustomerStatusFromJSONTyped = EndCustomerStatusFromJSONTyped;
function EndCustomerStatusToJSON(value) {
    return value;
}
exports.EndCustomerStatusToJSON = EndCustomerStatusToJSON;
