// xendit-node/customer/models/GetCustomerByReferenceID400ResponseAllOf.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetCustomerByReferenceID400ResponseAllOfToJSON = exports.GetCustomerByReferenceID400ResponseAllOfFromJSONTyped = exports.GetCustomerByReferenceID400ResponseAllOfFromJSON = exports.instanceOfGetCustomerByReferenceID400ResponseAllOf = exports.GetCustomerByReferenceID400ResponseAllOfErrorCodeEnum = void 0;
const runtime_1 = require("../../runtime");
/**
 * @export
 */
exports.GetCustomerByReferenceID400ResponseAllOfErrorCodeEnum = {
    EntityNotFoundError: 'ENTITY_NOT_FOUND_ERROR',
    ClientNotFoundError: 'CLIENT_NOT_FOUND_ERROR',
    EndCustomerNotFoundError: 'END_CUSTOMER_NOT_FOUND_ERROR',
    ApiValidationError: 'API_VALIDATION_ERROR'
};
/**
 * Check if a given object implements the GetCustomerByReferenceID400ResponseAllOf interface.
 */
function instanceOfGetCustomerByReferenceID400ResponseAllOf(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfGetCustomerByReferenceID400ResponseAllOf = instanceOfGetCustomerByReferenceID400ResponseAllOf;
function GetCustomerByReferenceID400ResponseAllOfFromJSON(json) {
    return GetCustomerByReferenceID400ResponseAllOfFromJSONTyped(json, false);
}
exports.GetCustomerByReferenceID400ResponseAllOfFromJSON = GetCustomerByReferenceID400ResponseAllOfFromJSON;
function GetCustomerByReferenceID400ResponseAllOfFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'errorCode': !(0, runtime_1.exists)(json, 'error_code') ? undefined : json['error_code'],
        'message': !(0, runtime_1.exists)(json, 'message') ? undefined : json['message'],
    };
}
exports.GetCustomerByReferenceID400ResponseAllOfFromJSONTyped = GetCustomerByReferenceID400ResponseAllOfFromJSONTyped;
function GetCustomerByReferenceID400ResponseAllOfToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'error_code': value.errorCode,
        'message': value.message,
    };
}
exports.GetCustomerByReferenceID400ResponseAllOfToJSON = GetCustomerByReferenceID400ResponseAllOfToJSON;
