// xendit-node/customer/models/GetCustomerByReferenceID400Response.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetCustomerByReferenceID400ResponseToJSON = exports.GetCustomerByReferenceID400ResponseFromJSONTyped = exports.GetCustomerByReferenceID400ResponseFromJSON = exports.instanceOfGetCustomerByReferenceID400Response = exports.GetCustomerByReferenceID400ResponseErrorCodeEnum = void 0;
const runtime_1 = require("../../runtime");
/**
 * @export
 */
exports.GetCustomerByReferenceID400ResponseErrorCodeEnum = {
    EntityNotFoundError: 'ENTITY_NOT_FOUND_ERROR',
    ClientNotFoundError: 'CLIENT_NOT_FOUND_ERROR',
    EndCustomerNotFoundError: 'END_CUSTOMER_NOT_FOUND_ERROR',
    ApiValidationError: 'API_VALIDATION_ERROR'
};
/**
 * Check if a given object implements the GetCustomerByReferenceID400Response interface.
 */
function instanceOfGetCustomerByReferenceID400Response(value) {
    let isInstance = true;
    isInstance = isInstance && "errorCode" in value;
    isInstance = isInstance && "message" in value;
    return isInstance;
}
exports.instanceOfGetCustomerByReferenceID400Response = instanceOfGetCustomerByReferenceID400Response;
function GetCustomerByReferenceID400ResponseFromJSON(json) {
    return GetCustomerByReferenceID400ResponseFromJSONTyped(json, false);
}
exports.GetCustomerByReferenceID400ResponseFromJSON = GetCustomerByReferenceID400ResponseFromJSON;
function GetCustomerByReferenceID400ResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'errorCode': json['error_code'],
        'message': json['message'],
        'errors': !(0, runtime_1.exists)(json, 'errors') ? undefined : json['errors'],
    };
}
exports.GetCustomerByReferenceID400ResponseFromJSONTyped = GetCustomerByReferenceID400ResponseFromJSONTyped;
function GetCustomerByReferenceID400ResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'error_code': value.errorCode,
        'message': value.message,
        'errors': value.errors,
    };
}
exports.GetCustomerByReferenceID400ResponseToJSON = GetCustomerByReferenceID400ResponseToJSON;
