// xendit-node/customer/models/IdentityAccountType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IdentityAccountTypeToJSON = exports.IdentityAccountTypeFromJSONTyped = exports.IdentityAccountTypeFromJSON = exports.IdentityAccountType = void 0;
/**
 *
 * @export
 */
exports.IdentityAccountType = {
    BankAccount: 'BANK_ACCOUNT',
    Ewallet: 'EWALLET',
    CreditCard: 'CREDIT_CARD',
    PayLater: 'PAY_LATER',
    Otc: 'OTC',
    QrCode: 'QR_CODE',
    XenditEnumDefaultFallback: "UNKNOWN_ENUM_VALUE"
};
function IdentityAccountTypeFromJSON(json) {
    return IdentityAccountTypeFromJSONTyped(json, false);
}
exports.IdentityAccountTypeFromJSON = IdentityAccountTypeFromJSON;
function IdentityAccountTypeFromJSONTyped(json, ignoreDiscriminator) {
    if (json !== "" || json !== null) {
        const key = Object.keys(exports.IdentityAccountType)[Object.values(exports.IdentityAccountType).indexOf(json)];
        return exports.IdentityAccountType[key] === undefined ?
            exports.IdentityAccountType['XenditEnumDefaultFallback'] : exports.IdentityAccountType[key];
    }
    return json;
}
exports.IdentityAccountTypeFromJSONTyped = IdentityAccountTypeFromJSONTyped;
function IdentityAccountTypeToJSON(value) {
    return value;
}
exports.IdentityAccountTypeToJSON = IdentityAccountTypeToJSON;
