// xendit-node/customer/models/IndividualDetail.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IndividualDetailToJSON = exports.IndividualDetailFromJSONTyped = exports.IndividualDetailFromJSON = exports.instanceOfIndividualDetail = exports.IndividualDetailGenderEnum = void 0;
const runtime_1 = require("../../runtime");
const EmploymentDetail_1 = require("./EmploymentDetail");
/**
 * @export
 */
exports.IndividualDetailGenderEnum = {
    Male: 'MALE',
    Female: 'FEMALE',
    Other: 'OTHER'
};
/**
 * Check if a given object implements the IndividualDetail interface.
 */
function instanceOfIndividualDetail(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfIndividualDetail = instanceOfIndividualDetail;
function IndividualDetailFromJSON(json) {
    return IndividualDetailFromJSONTyped(json, false);
}
exports.IndividualDetailFromJSON = IndividualDetailFromJSON;
function IndividualDetailFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'givenNames': !(0, runtime_1.exists)(json, 'given_names') ? undefined : json['given_names'],
        'givenNamesNonRoman': !(0, runtime_1.exists)(json, 'given_names_non_roman') ? undefined : json['given_names_non_roman'],
        'middleName': !(0, runtime_1.exists)(json, 'middle_name') ? undefined : json['middle_name'],
        'surname': !(0, runtime_1.exists)(json, 'surname') ? undefined : json['surname'],
        'surnameNonRoman': !(0, runtime_1.exists)(json, 'surname_non_roman') ? undefined : json['surname_non_roman'],
        'motherMaidenName': !(0, runtime_1.exists)(json, 'mother_maiden_name') ? undefined : json['mother_maiden_name'],
        'gender': !(0, runtime_1.exists)(json, 'gender') ? undefined : json['gender'],
        'dateOfBirth': !(0, runtime_1.exists)(json, 'date_of_birth') ? undefined : json['date_of_birth'],
        'nationality': !(0, runtime_1.exists)(json, 'nationality') ? undefined : json['nationality'],
        'placeOfBirth': !(0, runtime_1.exists)(json, 'place_of_birth') ? undefined : json['place_of_birth'],
        'employment': !(0, runtime_1.exists)(json, 'employment') ? undefined : (0, EmploymentDetail_1.EmploymentDetailFromJSON)(json['employment']),
    };
}
exports.IndividualDetailFromJSONTyped = IndividualDetailFromJSONTyped;
function IndividualDetailToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'given_names': value.givenNames,
        'given_names_non_roman': value.givenNamesNonRoman,
        'middle_name': value.middleName,
        'surname': value.surname,
        'surname_non_roman': value.surnameNonRoman,
        'mother_maiden_name': value.motherMaidenName,
        'gender': value.gender,
        'date_of_birth': value.dateOfBirth,
        'nationality': value.nationality,
        'place_of_birth': value.placeOfBirth,
        'employment': (0, EmploymentDetail_1.EmploymentDetailToJSON)(value.employment),
    };
}
exports.IndividualDetailToJSON = IndividualDetailToJSON;
