// xendit-node/customer/models/KYCDocumentType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.KYCDocumentTypeToJSON = exports.KYCDocumentTypeFromJSONTyped = exports.KYCDocumentTypeFromJSON = exports.KYCDocumentType = void 0;
/**
 *
 * @export
 */
exports.KYCDocumentType = {
    BirthCertificate: 'BIRTH_CERTIFICATE',
    BankStatement: 'BANK_STATEMENT',
    DrivingLicense: 'DRIVING_LICENSE',
    IdentityCard: 'IDENTITY_CARD',
    Passport: 'PASSPORT',
    Visa: 'VISA',
    BusinessRegistration: 'BUSINESS_REGISTRATION',
    BusinessLicense: 'BUSINESS_LICENSE',
    XenditEnumDefaultFallback: "UNKNOWN_ENUM_VALUE"
};
function KYCDocumentTypeFromJSON(json) {
    return KYCDocumentTypeFromJSONTyped(json, false);
}
exports.KYCDocumentTypeFromJSON = KYCDocumentTypeFromJSON;
function KYCDocumentTypeFromJSONTyped(json, ignoreDiscriminator) {
    if (json !== "" || json !== null) {
        const key = Object.keys(exports.KYCDocumentType)[Object.values(exports.KYCDocumentType).indexOf(json)];
        return exports.KYCDocumentType[key] === undefined ?
            exports.KYCDocumentType['XenditEnumDefaultFallback'] : exports.KYCDocumentType[key];
    }
    return json;
}
exports.KYCDocumentTypeFromJSONTyped = KYCDocumentTypeFromJSONTyped;
function KYCDocumentTypeToJSON(value) {
    return value;
}
exports.KYCDocumentTypeToJSON = KYCDocumentTypeToJSON;
