// xendit-node/customer/models/KYCDocumentSubType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.KYCDocumentSubTypeToJSON = exports.KYCDocumentSubTypeFromJSONTyped = exports.KYCDocumentSubTypeFromJSON = exports.KYCDocumentSubType = void 0;
/**
 *
 * @export
 */
exports.KYCDocumentSubType = {
    NationalId: 'NATIONAL_ID',
    ConsularId: 'CONSULAR_ID',
    VoterId: 'VOTER_ID',
    PostalId: 'POSTAL_ID',
    ResidencePermit: 'RESIDENCE_PERMIT',
    TaxId: 'TAX_ID',
    StudentId: 'STUDENT_ID',
    MilitaryId: 'MILITARY_ID',
    MedicalId: 'MEDICAL_ID',
    Others: 'OTHERS',
    XenditEnumDefaultFallback: "UNKNOWN_ENUM_VALUE"
};
function KYCDocumentSubTypeFromJSON(json) {
    return KYCDocumentSubTypeFromJSONTyped(json, false);
}
exports.KYCDocumentSubTypeFromJSON = KYCDocumentSubTypeFromJSON;
function KYCDocumentSubTypeFromJSONTyped(json, ignoreDiscriminator) {
    if (json !== "" || json !== null) {
        const key = Object.keys(exports.KYCDocumentSubType)[Object.values(exports.KYCDocumentSubType).indexOf(json)];
        return exports.KYCDocumentSubType[key] === undefined ?
            exports.KYCDocumentSubType['XenditEnumDefaultFallback'] : exports.KYCDocumentSubType[key];
    }
    return json;
}
exports.KYCDocumentSubTypeFromJSONTyped = KYCDocumentSubTypeFromJSONTyped;
function KYCDocumentSubTypeToJSON(value) {
    return value;
}
exports.KYCDocumentSubTypeToJSON = KYCDocumentSubTypeToJSON;
