// xendit-node/customer/models/ResponseDataNotFound.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResponseDataNotFoundToJSON = exports.ResponseDataNotFoundFromJSONTyped = exports.ResponseDataNotFoundFromJSON = exports.instanceOfResponseDataNotFound = exports.ResponseDataNotFoundErrorCodeEnum = void 0;
const runtime_1 = require("../../runtime");
/**
 * @export
 */
exports.ResponseDataNotFoundErrorCodeEnum = {
    DataNotFound: 'DATA_NOT_FOUND',
    EntityNotFoundError: 'ENTITY_NOT_FOUND_ERROR',
    ClientNotFoundError: 'CLIENT_NOT_FOUND_ERROR'
};
/**
 * Check if a given object implements the ResponseDataNotFound interface.
 */
function instanceOfResponseDataNotFound(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfResponseDataNotFound = instanceOfResponseDataNotFound;
function ResponseDataNotFoundFromJSON(json) {
    return ResponseDataNotFoundFromJSONTyped(json, false);
}
exports.ResponseDataNotFoundFromJSON = ResponseDataNotFoundFromJSON;
function ResponseDataNotFoundFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'errorCode': !(0, runtime_1.exists)(json, 'error_code') ? undefined : json['error_code'],
        'message': !(0, runtime_1.exists)(json, 'message') ? undefined : json['message'],
    };
}
exports.ResponseDataNotFoundFromJSONTyped = ResponseDataNotFoundFromJSONTyped;
function ResponseDataNotFoundToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'error_code': value.errorCode,
        'message': value.message,
    };
}
exports.ResponseDataNotFoundToJSON = ResponseDataNotFoundToJSON;
