// xendit-node/customer/models/UpdateCustomer400Response.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateCustomer400ResponseToJSON = exports.UpdateCustomer400ResponseFromJSONTyped = exports.UpdateCustomer400ResponseFromJSON = exports.instanceOfUpdateCustomer400Response = exports.UpdateCustomer400ResponseErrorCodeEnum = void 0;
const runtime_1 = require("../../runtime");
/**
 * @export
 */
exports.UpdateCustomer400ResponseErrorCodeEnum = {
    EntityNotFoundError: 'ENTITY_NOT_FOUND_ERROR',
    ClientNotFoundError: 'CLIENT_NOT_FOUND_ERROR',
    EndCustomerNotFoundError: 'END_CUSTOMER_NOT_FOUND_ERROR',
    DuplicateEndCustomerError: 'DUPLICATE_END_CUSTOMER_ERROR',
    ApiValidationError: 'API_VALIDATION_ERROR'
};
/**
 * Check if a given object implements the UpdateCustomer400Response interface.
 */
function instanceOfUpdateCustomer400Response(value) {
    let isInstance = true;
    isInstance = isInstance && "errorCode" in value;
    isInstance = isInstance && "message" in value;
    return isInstance;
}
exports.instanceOfUpdateCustomer400Response = instanceOfUpdateCustomer400Response;
function UpdateCustomer400ResponseFromJSON(json) {
    return UpdateCustomer400ResponseFromJSONTyped(json, false);
}
exports.UpdateCustomer400ResponseFromJSON = UpdateCustomer400ResponseFromJSON;
function UpdateCustomer400ResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'errorCode': json['error_code'],
        'message': json['message'],
        'errors': !(0, runtime_1.exists)(json, 'errors') ? undefined : json['errors'],
    };
}
exports.UpdateCustomer400ResponseFromJSONTyped = UpdateCustomer400ResponseFromJSONTyped;
function UpdateCustomer400ResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'error_code': value.errorCode,
        'message': value.message,
        'errors': value.errors,
    };
}
exports.UpdateCustomer400ResponseToJSON = UpdateCustomer400ResponseToJSON;
