// xendit-node/customer/models/UpdateCustomer400ResponseAllOf.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateCustomer400ResponseAllOfToJSON = exports.UpdateCustomer400ResponseAllOfFromJSONTyped = exports.UpdateCustomer400ResponseAllOfFromJSON = exports.instanceOfUpdateCustomer400ResponseAllOf = exports.UpdateCustomer400ResponseAllOfErrorCodeEnum = void 0;
const runtime_1 = require("../../runtime");
/**
 * @export
 */
exports.UpdateCustomer400ResponseAllOfErrorCodeEnum = {
    EntityNotFoundError: 'ENTITY_NOT_FOUND_ERROR',
    ClientNotFoundError: 'CLIENT_NOT_FOUND_ERROR',
    EndCustomerNotFoundError: 'END_CUSTOMER_NOT_FOUND_ERROR',
    DuplicateEndCustomerError: 'DUPLICATE_END_CUSTOMER_ERROR',
    ApiValidationError: 'API_VALIDATION_ERROR'
};
/**
 * Check if a given object implements the UpdateCustomer400ResponseAllOf interface.
 */
function instanceOfUpdateCustomer400ResponseAllOf(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfUpdateCustomer400ResponseAllOf = instanceOfUpdateCustomer400ResponseAllOf;
function UpdateCustomer400ResponseAllOfFromJSON(json) {
    return UpdateCustomer400ResponseAllOfFromJSONTyped(json, false);
}
exports.UpdateCustomer400ResponseAllOfFromJSON = UpdateCustomer400ResponseAllOfFromJSON;
function UpdateCustomer400ResponseAllOfFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'errorCode': !(0, runtime_1.exists)(json, 'error_code') ? undefined : json['error_code'],
        'message': !(0, runtime_1.exists)(json, 'message') ? undefined : json['message'],
    };
}
exports.UpdateCustomer400ResponseAllOfFromJSONTyped = UpdateCustomer400ResponseAllOfFromJSONTyped;
function UpdateCustomer400ResponseAllOfToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'error_code': value.errorCode,
        'message': value.message,
    };
}
exports.UpdateCustomer400ResponseAllOfToJSON = UpdateCustomer400ResponseAllOfToJSON;
