// xendit-node/payout/models/ChannelCategory.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChannelCategoryToJSON = exports.ChannelCategoryFromJSONTyped = exports.ChannelCategoryFromJSON = exports.ChannelCategory = void 0;
/**
 * Category of channel code, as some channels might require more fields during processing
 * @export
 */
exports.ChannelCategory = {
    Bank: 'BANK',
    Ewallet: 'EWALLET',
    Otc: 'OTC',
    XenditEnumDefaultFallback: "UNKNOWN_ENUM_VALUE"
};
function ChannelCategoryFromJSON(json) {
    return ChannelCategoryFromJSONTyped(json, false);
}
exports.ChannelCategoryFromJSON = ChannelCategoryFromJSON;
function ChannelCategoryFromJSONTyped(json, ignoreDiscriminator) {
    if (json !== "" || json !== null) {
        const key = Object.keys(exports.ChannelCategory)[Object.values(exports.ChannelCategory).indexOf(json)];
        return exports.ChannelCategory[key] === undefined ?
            exports.ChannelCategory['XenditEnumDefaultFallback'] : exports.ChannelCategory[key];
    }
    return json;
}
exports.ChannelCategoryFromJSONTyped = ChannelCategoryFromJSONTyped;
function ChannelCategoryToJSON(value) {
    return value;
}
exports.ChannelCategoryToJSON = ChannelCategoryToJSON;
