// xendit-node/payout/models/ChannelAccountType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChannelAccountTypeToJSON = exports.ChannelAccountTypeFromJSONTyped = exports.ChannelAccountTypeFromJSON = exports.ChannelAccountType = void 0;
/**
 * Available account types (applicable for MY_DUITNOW)
 * @export
 */
exports.ChannelAccountType = {
    NationalId: 'NATIONAL_ID',
    MobileNo: 'MOBILE_NO',
    Passport: 'PASSPORT',
    BusinessRegistration: 'BUSINESS_REGISTRATION',
    BankAccount: 'BANK_ACCOUNT',
    XenditEnumDefaultFallback: "UNKNOWN_ENUM_VALUE"
};
function ChannelAccountTypeFromJSON(json) {
    return ChannelAccountTypeFromJSONTyped(json, false);
}
exports.ChannelAccountTypeFromJSON = ChannelAccountTypeFromJSON;
function ChannelAccountTypeFromJSONTyped(json, ignoreDiscriminator) {
    if (json !== "" || json !== null) {
        const key = Object.keys(exports.ChannelAccountType)[Object.values(exports.ChannelAccountType).indexOf(json)];
        return exports.ChannelAccountType[key] === undefined ?
            exports.ChannelAccountType['XenditEnumDefaultFallback'] : exports.ChannelAccountType[key];
    }
    return json;
}
exports.ChannelAccountTypeFromJSONTyped = ChannelAccountTypeFromJSONTyped;
function ChannelAccountTypeToJSON(value) {
    return value;
}
exports.ChannelAccountTypeToJSON = ChannelAccountTypeToJSON;
